src/main/java/repo/impl/PostgresMigratedFormSessionRepo.java

```diff
@@ -0,0 +1,416 @@
+        }
+
+        public void setAppId(String appId) {
+            this.appId = appId;
+        }
+
+        public String getSessionName() {
+            return sessionName;
+        }
+
+        public void setSessionName(String sessionName) {
+            this.sessionName = sessionName;
+        }
+
+        public String getCreatedDate() {
+            return createdDate;
+        }
+
+        public void setCreatedDate(String createdDate) {
+            this.createdDate = createdDate;
+        }
+
+        public String getLastModified() {
+            return lastModified;
+        }
+
+        public void setLastModified(String lastModified) {
+            this.lastModified = lastModified;
+        }
+
+        public String getUserId() {
+            return userId;
+        }
+
+        public void setUserId(String userId) {
+            this.userId = userId;
+        }
+    }
+
+
+    public static SerializableFormSession loadSessionFromJson(String sessionJSON) {
+        SerializableFormSession session = new SerializableFormSession();
+        JSONObject sessionObject = new JSONObject(sessionJSON);
+        JSONObject sessionData = sessionObject.getJSONObject("session_data");
+        session.setTitle(sessionData.getString("session_name"));
+        session.setUsername(sessionData.getString("username"));
+        session.setDomain(sessionData.getString("domain"));
+        session.setOneQuestionPerScreen(false);
+        session.setAsUser(null);
+        session.setInstanceXml(sessionObject.getString("instance"));
+        session.setFormXml(sessionObject.getString("xform"));
+        session.setInitLang(sessionObject.getString("init_lang"));
+        session.setSequenceId(sessionObject.getInt("seq_id"));
+        HashMap<String, String> sessionDataMap = new HashMap<>();
+        session.setSessionData(sessionDataMap);
+        if (sessionData.has("case_id")) {
+            sessionDataMap.put("case_id", (String) sessionData.get("case_id"));
+        }
+        return session;
+    }
+
+    // Copied from FormSession
+    private static String serializeFormDef(FormDef formDef) {
+        try {
+            org.apache.commons.io.output.ByteArrayOutputStream baos = new org.apache.commons.io.output.ByteArrayOutputStream();
+            DataOutputStream serializedStream = new DataOutputStream(baos);
+            formDef.writeExternal(serializedStream);
+            return Base64.encodeBase64String(baos.toByteArray());
+        } catch(Exception e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+    // Copied from NewFormResponseFactory
+    private static FormDef parseFormDef(String formXml) {
+        try {
+            XFormParser mParser = new XFormParser(new StringReader(formXml));
+            return mParser.parse();
+        } catch(Exception e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    private static String convertIsoToJavaDate(String date) {
```

Comment:
it's counter part lives here: https://github.com/dimagi/formplayer/blob/master/src/main/java/util/FormplayerDateUtils.java#L14. may make sense to put them in the same spot